src/components/MainSidebar/TimeDimensions.js

```diff
@@ -22,6 +22,9 @@ import { PROGRAM_TYPE_WITH_REGISTRATION } from './ProgramDimensionsPanel/Program
 import styles from './TimeDimensions.module.css'
 
 const getName = (dimension, program, stage) => {
+    if (!dimension.nameParentProperty) {
```

Comment:
This makes it much easier to follow, although probably it would work anyway if the metaData is fixed.

Reply:
Yes, I just figures since `LastUpdated` doesn't have a custom name property, that it is more correct to do it this way, even though the result is the same.